### PR TITLE
fix(carousel): swipe functionality

### PR DESCRIPTION
**Pull request by NSDrowned**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5029

### Description

Added swipe functionality, which includes 2 settings:
- `allowedTime`: min time allowed to do a swipe gesture (set to 300ms)
- `threshold`: min distance needed the gesture needs to travel to be considered a swipe (set to 75px)

Tested on Safari, Chrome and Firefox for iPhone.

### Changelog

**New**

- Added `handleTouchStartEvent`  and `handleTouchEndEvent` triggered by touchstart and touchend respectively

**Changed**

- Added same fix as #5030 to story to avoid the horizontal scroll bug that makes the new swipe functionality look like it's malfunctioning.

### CODE REVIEW

File: packages/web-components/src/components/carousel/carousel.ts [hunk 1]
```diff
@@ -179,6 +191,35 @@ class DDSCarousel extends HostListenerMixin(LitElement) {
...
+  }
+
+  /**
+   * Handles `touchend` event.
+   */
+  private _handleTouchEndEvent(event: TouchEvent) {
+    const { _startPos, _startTime } = this;
+    const { pageSize, start, _total: total } = this;
+    const allowedTime = 300; // max time allowed to do swipe
+    const threshold = 75; // min distance traveled to be considered swipe
```
Comment (asudoh) [inline packages/web-components/src/components/carousel/carousel.ts:209]: Should put these constants in private class fields or out-of-the class variables.

**Review by asudoh — COMMENTED** [review]
Thank you for working on this @NSDrowned!

File: packages/web-components/src/components/carousel/carousel.ts (hunk 1 above)
Comment (NSDrowned) [reply packages/web-components/src/components/carousel/carousel.ts]: Done =)

**Review by asudoh — COMMENTED** [review]
LGTM 👍 code wise, wanted to have designers see what they think. Thanks @NSDrowned!

**Review by shixiedesign — APPROVED** [review]
Works on Web components

**Review by shixiedesign — COMMENTED** [review]
Swiping works on Web components, but not working on React wrapper. If this is expected then we are all good!

**Review by jeffchew — APPROVED** [review]
Thank you @NSDrowned !

**Pull request merged**